corrections and clarifications to Jakarta Validation section (#231)
jakartaee/data: spec/src/main/asciidoc/jakarta-ee.adoc

In this example, the `name` field is annotated with `@NotBlank`, indicating that it must not be blank. The `age` field is annotated with both `@Positive` and `@Min(18)`, ensuring it is a positive integer greater than or equal to 18.

> The `School` repository interface, shown below, uses the `jakarta.validation.Valid` annotation to cause the constraints from the `Student` entity to be applied to the `save` operation parameter, whereas the validation constraints are not applied to the `Student` entities that are the results of the `findByAgeLessThanEqual` operation because the example does not include any `jakarta.validation.Valid` annotation that applies to the return value.

Reviewer: ```suggestion
The `School` repository interface, shown below, uses the `jakarta.validation.Valid` annotation to cause the constraints from the `Student` entity to be validated during the `save` operation, whereas the validation constraints are not applied to the `Student` entities returned as a result of the `findByAgeLessThanEqual` operation because the `findByAgeLessThanEqual` method does not include a `jakarta.validation.Valid` annotation that applies to the return value.
```

[source,java]
----
@Repository

> public interface School extends DataRepository<Student, String> {

Reviewer 2: ```suggestion
public interface School extends DataRepository<@jakarta.validation.Valid Student, String> {
```
The way that Micronaut does sounds more semantic:

https://micronaut-projects.github.io/micronaut-data/latest/guide/

I mean, overwrite the method to include the valid annotation sounds noise in the method.

Author: > The way that Micronaut does sounds more semantic:

@otaviojava  I like the approach that Micronaut has as well.  But Jakarta Validation doesn't currently allow that pattern.  One of the Micronaut developers and myself have been in contact with the Jakarta Validation team about it, and I've opened issue https://github.com/jakartaee/validation/issues/186 officially requesting them to add this ability for Jakarta EE 11.  If they add it, we'll update our spec documentation to favor that approach.  But for now, our specification is limited to what is currently supported by Jakarta Validation.

>     void save(@Valid Student s);

Reviewer 2: ```suggestion
````

Reviewer 2: If this makes sense the semantic, I would remove this line.

Author: See explanation in response to previous comment.

Reviewer 2: Thanks.

    List<Student> findByAgeLessThanEqual(@Min(18) int age);
}
----

==== Avoiding Overlap with Validation from Jakarta Persistence

> Jakarta Data providers that are built using Jakarta Persistence might require the user to define persistence units for repositories or might handle the details of defining the persistence units internally. A user that defines the persistence unit for a Jakarta Data repository specifies the `validation-mode` as `NONE` per the "Enabling Automatic Validation" section of the Jakarta Persistence specification to avoid duplicate validation of entities. Similarly, the Jakarta Data provider uses the `validation-mode` of `NONE` or the `jakarta.persistence.validation.mode` map key with value of `none` that is defined in the "Enabling Automatic Validation" section of the Jakarta Persistence specification to avoid duplicate validation of entities.

Reviewer: ```suggestion
Jakarta Data providers that are built using Jakarta Persistence might require the user to define persistence units for repositories or might handle the details of defining the persistence units internally. A user that defines the persistence unit for a Jakarta Data repository must specify the `validation-mode` as `NONE` per the "Enabling Automatic Validation" section of the Jakarta Persistence specification to avoid duplicate validation of entities. Similarly, the Jakarta Data provider must specify either the `validation-mode` of `NONE` or the `jakarta.persistence.validation.mode` map key with value of `none` that is defined in the "Enabling Automatic Validation" section of the Jakarta Persistence specification to avoid duplicate validation of entities.
```
